Automattic/simplenote-ios
# Use API calls to trigger release jobs

### Fix
This PR updates our release tools to trigger any CI release job via an API call instead of relying on the automatic trigger on tags.
This way, it will be easier and cleaner to handle any failure during the release build and, moreover, it'll be possible to move more steps of the releases to CI (for example, validation and lint steps that can fail by design).

Note that in order to use the release tools, users should configure a new token for CircleCI on their local environment, following these steps:

- Go to https://app.circleci.com/settings/user/tokens.
- Create a Personal API Token.
- Add `CIRCLE_CI_AUTH_TOKEN=<Your Token>` to your `~/.wpios-env.default` (you can check out the updated fastlane/env/user.env-example for a template).

This PR needs an updated version of `release-toolkit`, so it's now pointing to https://github.com/wordpress-mobile/release-toolkit/pull/223. After approval, I'll release a new version of `release-toolkit` and update this PR before merging.

### Test
Full testing is tricky because it's tight to the release cycle. Anyway, it's possible to test the critical part of it this way:

- checkout a new `release/test` branch from this one.
- add a dummy commit (the build will fail, so adding a dummy commit will avoid the failure to be reported to this PR). 
- `bundle install`
- Run `bundle exec fastlane trigger_beta_build branch_to_build:"release/test"`
- Verify that the `release build` job has been started on CircleCI. The job will fail during the upload to AppStore because the current version is already present. This is fine. 
- Delete the test branch.

### Review
I asked the review to @mokagio because he is currently the release manager of Simplenote iOS. All the code is already in production in WPiOS, so it _should_ be fine, but there may always be some nuances with porting that I missed, so I think that understanding the changes will help if something happens during the next release. 

### Release
These changes do not require release notes.
